### ISSUE TITLE
Enhancement request - versioned plugin documentation

### ISSUE THREAD

**Issue by kript** [irods/irods_docs#142]
Hi folks,

I find having the documentation versioned incredibly helpful and useful. I constantly refer to it and send pertinent links to colleagues. 

Can we extend this to the plugins please, so that each set of documentation also includes the plugins for each version, or the plugins have a similar interface?

**Comment by korydraughn** [comment]
Are you saying you want a section in the docs that links to the correct version tag for each plugin repository?

If not, how would you expect it to work?

**Comment by kript** [comment]
> Are you saying you want a section in the docs that links to the correct version tag for each plugin repository?
> 

While that would be ideal, I can see how it could be tricky to incorporate multiple versions of a plugin (not everyone can run the latest).

> If not, how would you expect it to work?

Another thought would be to have e.g. docs.irods.org/plugins/indexing/4.2.7.1 which I naively think would be easier to generate?

**Comment by kript** [comment]
> I think you can just... link to the tagged version at GitHub?

Ideally, it would be a readthedocs/Doxygen style similar to docs.irods.org, rather than a collection of links, even though that would be the simplest. 

I'd argue that you would want to move the docs for the plugins there in the same way the documentation for iRODS itself is held in a different repo.

The overall idea is that there is a one-stop place for all the documentation, which is added on every release and therefore also becomes also a catalogue of available options, rather than admins looking through the GitHub organisation and seeing if their version is supported.

**Comment by trel** [comment]
So we would... copy/collect/gather markdown from multiple other repositories (the plugins), and make it part of the buildout of the main source that gets generated/pushed to docs.irods.org.   That seems, possible, if not a bit brittle (requires things always continuing to work in lockstep).

But, since the plugins move at their own pace... point releases of the plugins would not necessarily get added to main releases of the server.   Will have to keep thinking on how that would/could work.

**Comment by kript** [comment]
> But, since the plugins move at their own pace... point releases of the plugins would not necessarily get added to main releases of the server. Will have to keep thinking on how that would/could work.

AIUI the documentation is in its own repository, so updates could be made to that every time a version of a plugin is released?

**Comment by trel** [comment]
Right, but the documentation for each plugin live in each plugin repository.

Oh... wait... you're suggesting it NOT live in each plugin repository?   Consolidate all-the-things....

**Comment by kript** [comment]
Yes, exactly. Then you get the side benefit of URL-based versioned documentation for each plugin on the same site that the rest of the documentation of the project is held on.
Thus making it a one-stop place for all the documentation and an easy way for new people to get an idea of the 'ecosystem'.
